otherlibs/dune-rpc/private/exported_types.ml

```diff
@@ -34,6 +34,402 @@ module Loc = struct
   ;;
 end
 
+module Ansi_color = struct
+  module RGB8 = struct
+    type t = char
+
+    let sexp = Conv.char
+    let to_ansi_color_rgb8 t = Stdune.Ansi_color.RGB8.of_char t
+    let of_ansi_color_rgb8 t = Stdune.Ansi_color.RGB8.to_char t
+  end
+
+  module RGB24 = struct
+    type t = int
+
+    let sexp = Conv.int
+    let to_ansi_color_rgb24 t = Stdune.Ansi_color.RGB24.of_int t
+    let of_ansi_color_rgb24 t = Stdune.Ansi_color.RGB24.to_int t
+  end
+
+  module Style = struct
+    type t =
+      | Fg_default
+      | Fg_black
+      | Fg_red
+      | Fg_green
+      | Fg_yellow
+      | Fg_blue
+      | Fg_magenta
+      | Fg_cyan
+      | Fg_white
+      | Fg_bright_black
+      | Fg_bright_red
+      | Fg_bright_green
+      | Fg_bright_yellow
+      | Fg_bright_blue
+      | Fg_bright_magenta
+      | Fg_bright_cyan
+      | Fg_bright_white
+      | Fg_8_bit_color of RGB8.t
+      | Fg_24_bit_color of RGB24.t
+      | Bg_default
+      | Bg_black
+      | Bg_red
+      | Bg_green
+      | Bg_yellow
+      | Bg_blue
+      | Bg_magenta
+      | Bg_cyan
+      | Bg_white
+      | Bg_bright_black
+      | Bg_bright_red
+      | Bg_bright_green
+      | Bg_bright_yellow
+      | Bg_bright_blue
+      | Bg_bright_magenta
+      | Bg_bright_cyan
+      | Bg_bright_white
+      | Bg_8_bit_color of RGB8.t
+      | Bg_24_bit_color of RGB24.t
+      | Bold
+      | Dim
+      | Italic
+      | Underline
+
+    let sexp =
+      let open Conv in
```

Comment:
Could you introduce `Conv.enum` to make this all a bit more compact?

Reply:
`Conv.enum` as it exists is not sufficient since I cannot handle constructor arguments.